test/distributed/test_functional_api.py

```diff
@@ -427,6 +426,11 @@ def allred_mesh_dim(input):
 WORLD_SIZE = 2
 
 
+def exit_if_lt_x_gpu(x):
+    if BACKEND == dist.Backend.NCCL and torch.cuda.device_count() < x:
+        sys.exit(TEST_SKIPS[f"multi-gpu-{x}"].exit_code)
```

Comment:
Is `sys.exit` to skip the test or exit the test suit completely?  As I see it is used in `test_tracing_with_fakepg`, I assume it would be the former (to skip that test).  Is that better to use https://docs.python.org/3/library/unittest.html#unittest.SkipTest instead of `sys.exit`

Reply:
Oh, I now see @malfet comment above.

Reply:
For distributed tests, sys.exit is used.  See the decorator `skip_if_lt_x_gpu` in `torch/testing/_internal/common_distributed.py`.

Reply:
Can someone else experiment with not using sys.exit in a follow-up PR instead of having me do it here?  I'm trying to get our distributed CI signal back to green and I'm following the precedent established elsewhere in this file and elsewhere in torch/testing/_internal.

Reply:
Well, my problem with that decorator, is that we can skip a bunch of tests because we exit from the testsuite early?

As a middle ground can we check that exit code is non-zero here? Or should it be?

Reply:
yea we shouldn't do exit here.  why is the skip_if_lt_x_gpu macro not working here?

Reply:
See my previous comment https://github.com/pytorch/pytorch/pull/116217#issuecomment-1867012873.